docs/schemaform/form-config.md

```diff
@@ -0,0 +1,297 @@
+    }
+  ],
+  
+  // An object with field specific error messages. Structured by error name (from JSON
+  // Schema error types). This is passed to custom validations in `ui:validations` if
+  // you want to allow configurable error messages in a validator.
+  'ui:errorMessages': {
+    errorType: ''
+  },
+  'ui:options': {
+    
+    // This is a string of class names that will be added to the widget for the current
+    // field. Similar to the default `classNames` property, but will put the class names
+    // on the input/select/etc element itself, rather than a surrounding `div`.
+    widgetClassNames: '', 
+    
+    // For Array fields, this is a component that is shown when the item in the array is
+    // being shown read-only on a normal form page (i.e. not on the review page).
+    viewField: RowViewComponent, 
+    
+    // If you want a field to only be shown when another field is true, set this option
+    // to the property name. It will follow our ExpandingGroup pattern and expand underneath
+    // the field it is set to.
+    expandUnder: '', 
+    
+    // Set this if you want to hide this field on the review page.
+    hideOnReview: true || false,
+
+    // Set this if you want to hide this field on the review page when the field value is falsy
+    hideOnReviewIfFalse: true || false 
+
+    // Function that conditionally hides fields in the form
+    hideIf: function (fieldData) {
+      return true || false;
+    }
+
+    // Function that conditionally replaces the current field's schema
+    updateSchema: function (fieldData, pageData) {
+      return {};
+    }
+  }
+}
+```
+
+## Schemaform cookbook
+
+Here are some common situations you might run into when building a form and how to address them.
+
+### I need to write custom validation
+
+JSON Schema does not provide all the validation options we need in our forms, so we've created an additional way to add field validations, using `ui:validations` in the uiSchema object. `ui:validations` is an array and each item can be a function or an object. If you pass a function, it will be called with the following arguments:
+
+- errors: The errors object for the field.
+- fieldData: The data for the field.
+- pageData: The current form (page) data.
+- schema: The current JSON Schema for the field.
+- errorMessages: The error messsage object (if available) for thie field.
+
+Every validation function should update the errors object with any errors found. This is done by calling its `addErrors()` method. Here's an example:
+
+```js
+function validateSSN(errors, ssn) {
+  if (!isValidSSN(ssn)) {
+    errors.addError('Please enter a valid nine digit SSN (dashes allowed)');
+  }
+}
+```
+
+Items in the `ui:validations` array can also be objects. Objects should have two properties:
+
+- options: Object (or anything, really) that will be passed to your validation function. You can use this to allow your validation function to be configurable for different fields on the form.
```

Comment:
I think an example for this would be useful, it's not super clear reading the docs how this is used.